students/DrewSmith/lessons/lesson06/assignment/good_perf.py

```diff
@@ -0,0 +1,70 @@
+"""
+good performing, perfectly written module :)
+
+"""
+
+
+import datetime
+from dateutil.parser import parse
+from datetime import datetime as dt
+
+def analyze(filename):
+    '''
+    Reads and analyzes data from the given file
+
+    :param filename: file path of file to analyze
+    '''
+    start = datetime.datetime.now()
+    
+    year_count = {
+            "2013": 0,
+            "2014": 0,
+            "2015": 0,
+            "2016": 0,
+            "2017": 0,
+            "2018": 0
+    }
+    found = 0
+    with open(filename) as csvfile:      
+        row = csvfile.readline()
+        # skip the first record
+        row = csvfile.readline()
+        while row:
+            comma_index = row.rfind(",")
```

Comment:
Excellent. You are the first person to question using the csv library, and it paid off. If you were really going to change this in production code, you would want to do some research into what you are losing by not using the csv library, and if the tradeoffs are worth it.